test/compare_with_wikidata/csv_from_url_test.rb

```diff
@@ -0,0 +1,85 @@
+require 'test_helper'
+
+describe 'CompareWithWikidata::CSVFromURL' do
+  def csv_from_url(url)
+    CompareWithWikidata::CSVFromURL.new(url).parsed
+  end
+
+  it 'should produce a sensible response from genuine CSV' do
+    stub_request(:get, 'http://example.com/real.csv').to_return(
+      body: %(heading A,"heading B",heading C\r\n1,2,3\r\n)
+    )
+    csv_from_url('http://example.com/real.csv').must_equal [
+      { heading_a: '1', heading_b: '2', heading_c: '3' },
+    ]
+  end
+
+  it 'should produce a helpful exception message on a 404 error' do
+    stub_request(:get, 'http://example.com/non-existent').to_return(
+      status: [404, 'No file found. Nothing at all.']
+    )
+    error = assert_raises Exception do
+      csv_from_url('http://example.com/non-existent')
+    end
+    error.message.must_equal 'There was an error fetching: http://example.com/non-existent - the error was: 404 Not Found'
+  end
+
+  it 'should produce a helpful exception message on a 404 error' do
+    stub_request(:get, 'http://example.com/errors').to_return(
+      status: [500, 'Our fault. Not your fault.']
+    )
+    error = assert_raises Exception do
```

Comment:
Again, I realise you've inherited this, and fixing it is quite possibly out of scope here, depending on whether we class this as "new" code or not, but the mix of test styles here is a little awkward. This would probably be better using `must_raise`.